pkg/serving/v1/client.go

```diff
@@ -378,6 +378,13 @@ func getBaseRevision(cl KnServingClient, service *servingv1.Service) (*servingv1
 
 // Delete a revision by name
 func (cl *knServingClient) DeleteRevision(name string, timeout time.Duration) error {
+	revision, err := cl.client.Revisions(cl.namespace).Get(name, v1.GetOptions{})
+	if err != nil {
+		return clienterrors.GetError(err)
+	}
+	if revision.GetDeletionTimestamp() != nil {
+		return fmt.Errorf("can't delete revision %s because it has been marked for deletion", name)
```

Comment:
I think we've consistently used `'%s'` in messages... Please check and follow other messages